src/tools/cephfs/cephfs-shell

```diff
@@ -949,7 +950,18 @@ class CephFSShell(Cmd):
                     cephfs.unlink(path)
                 except libcephfs.Error as e:
                     # NOTE: perhaps we need a better msg here
-                    set_exit_code_msg(msg=e)
+                    if e.get_error_code() == 2:
+                        set_exit_code_msg(e.get_error_code(),
+                                          "rm: failed to remove "
+                                          f"{path.decode('utf-8')}: "
+                                          "No such file or directory")
+                    elif e.get_error_code() == 21:
```

Comment:
Need to check for 23 instead?
```suggestion
                    elif e.get_error_code() == 23:
```

Reply:
Oh no! I just recalled that the error numbers will be remapped by set_exit_code_msg().